### PR TITLE
chore: small progress on equiv and list basics

### PR DESCRIPTION
**Pull request by arthurpaulino**
Babysteps!

### CODE REVIEW

File: Mathlib/Data/Equiv/Basic.lean [hunk 1]
```diff
@@ -20,11 +20,11 @@ open Function
...
-structure equiv (α : Sort u) (β : Sort v) :=
-(to_fun    : α → β)
-(inv_fun   : β → α)
-(left_inv  : left_inverse inv_fun to_fun)
-(right_inv : right_inverse inv_fun to_fun)
+structure equiv (α : Sort u) (β : Sort v) where
+  to_fun    : α → β
+  inv_fun   : β → α
+  left_inv  : left_inverse inv_fun to_fun
+  right_inv : right_inverse inv_fun to_fun
```
Comment (gebner) [inline Mathlib/Data/Equiv/Basic.lean:27]:
```suggestion
structure Equiv (α : Sort u) (β : Sort v) where
  toFun    : α → β
  invFun   : β → α
  leftInv  : left_inverse invFun toFun
  rightInv : right_inverse invFun toFun
```

We might change it to the correct naming convention while you're already looking at the file.

**Review by gebner — COMMENTED** [review]
Otherwise lgtm.

File: Mathlib/Data/Equiv/Basic.lean [hunk 2]
```diff
@@ -43,12 +43,20 @@ rfl
...
+
 def symm (e : α ≃ β) : β ≃ α := ⟨e.inv_fun, e.to_fun, e.right_inv, e.left_inv⟩
 
+def simps.symm_apply (e : α ≃ β) : β → α := e.symm
+
 def trans (e₁ : α ≃ β) (e₂ : β ≃ γ) : α ≃ γ :=
 ⟨e₂ ∘ (e₁ : α → β), e₁.symm ∘ (e₂.symm : γ → β),
   e₂.left_inv.comp e₁.left_inv, e₂.right_inv.comp e₁.right_inv⟩
 
+@[simp] theorem to_fun_as_coe (e : α ≃ β) : e.to_fun = e := rfl
```
Comment (gebner) [inline Mathlib/Data/Equiv/Basic.lean:56]: BTW, please check the linter output.  It says this theorem shouldn't be marked simp (if you hover over rfl you might see why).

File: Mathlib/Data/Equiv/Basic.lean [hunk 3]
```diff
@@ -43,12 +43,20 @@ rfl
 
 def refl (α) : α ≃ α := ⟨id, id, λ _ => rfl, λ _ => rfl⟩
 
+instance : Inhabited (α ≃ α) := ⟨equiv.refl α⟩
+
 def symm (e : α ≃ β) : β ≃ α := ⟨e.inv_fun, e.to_fun, e.right_inv, e.left_inv⟩
 
+def simps.symm_apply (e : α ≃ β) : β → α := e.symm
```
[the remark this answers is not in the document]
Comment (arthurpaulino) [reply Mathlib/Data/Equiv/Basic.lean:50]: True, it's not needed right now. I pulled it from `mathlib3`

**Pull request merged**